source/fe/fe_series_fourier.cc

```diff
@@ -179,16 +189,23 @@ namespace FESeries
   Fourier<dim, spacedim>::Fourier(
     const std::vector<unsigned int> &      n_coefficients_per_direction,
     const hp::FECollection<dim, spacedim> &fe_collection,
-    const hp::QCollection<dim> &           q_collection)
+    const hp::QCollection<dim> &           q_collection,
+    const unsigned int                     component)
     : n_coefficients_per_direction(n_coefficients_per_direction)
     , fe_collection(&fe_collection)
     , q_collection(q_collection)
     , fourier_transform_matrices(fe_collection.size())
+    , component(component)
```

Comment:
With the invalid value, this would
```suggestion
    , component(component != numbers::invalid_unsigned_int ? component : 0)
```
and then you would add the following assertion below:
```
  if (fe_collection.n_component() > 0)
    Assert (component != numbers::invalid_unsigned_int,
                ExcMessage ("For vector-valued problems, you need to explicitly specify for which vector "
                        "component you will want to do a Fourier decomposition by setting the 'component' "
                        "argument of this constructor.")
```

Reply:
It would make sense to give the constructor parameter and the class member different names to avoid ambiguity, e.g., `component_` for the former and `component` for the latter.